File: python/LeetCode1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict_nums = {num: idx for idx, num in enumerate(nums)}
        for idx, num in enumerate(nums):
            if dict_nums.get(num, -1) == idx:
                dict_nums.pop(num)
            if (target-num) in dict_nums:
                return [idx, dict_nums[target-num]]

File: cpp/LeetCode1.cpp
#include <unordered_map>
#include <vector>

class Solution {
    public:
        std::vector<int> twoSum(std::vector<int>& nums, int target) {
            std::unordered_map<int, int> seen;
            for (int i = 0; i < nums.size(); i++) {
                int wanted_value = target - nums[i];
                
                if (seen.find(wanted_value) != seen.end()) {
                    return {i, seen[wanted_value]};
                }
    
                seen[nums[i]] = i;
            }
    
            return {};
        }
    };